nxp_s32k/libuavcan/driver/src/s32k_libuavcan.cpp

```diff
@@ -0,0 +1,777 @@
+        if (S32K::g_frame_ISRbuffer[instance].size() <= S32K::Frame_Capacity)
+        {
+            /* Harvest the Message buffer, read of the control and status word locks the MB */
+
+            /* Get the raw DLC from the message buffer that received a frame */
+            std::uint32_t dlc_ISR_raw =
+                ((S32K::FlexCAN[instance]->RAMn[MB_index * S32K::MB_Size_Words]) & CAN_WMBn_CS_DLC_MASK) >>
+                CAN_WMBn_CS_DLC_SHIFT;
+
+            /* Create CAN::FrameDLC type variable from the raw dlc */
+            CAN::FrameDLC dlc_ISR = CAN::FrameDLC(dlc_ISR_raw);
+
+            /* Convert from dlc to data length in bytes */
+            std::uint8_t payloadLength_ISR = S32K_InterfaceGroup::FrameType::dlcToLength(dlc_ISR);
+
+            /* Get the id */
+            std::uint32_t id_ISR =
+                (S32K::FlexCAN[instance]->RAMn[MB_index * S32K::MB_Size_Words + 1]) & CAN_WMBn_ID_ID_MASK;
+
+            /* Array for harvesting the received frame's payload */
+            std::uint32_t data_ISR_word[(payloadLength_ISR >> 2) + std::min(1, (payloadLength_ISR & 0x3))];
+
+            /* Perform the harvesting of the payload, leveraging from native 32-bit transfers and since the FlexCAN
+             * expects the data to be in big-endian order, a byte swap is required from the little-endian
+             * transmission UAVCAN requirement */
+            for (std::uint8_t i = 0;
+                 i < (payloadLength_ISR >> 2) + std::min(1, static_cast<std::uint8_t>(payloadLength_ISR) & 0x3);
+                 i++)
+            {
+                REV_BYTES_32(S32K::FlexCAN[instance]->RAMn[MB_index * S32K::MB_Size_Words + S32K::MB_Data_Offset + i],
+                             data_ISR_word[i]);
+            }
+
+            /* Harvest the frame's 16-bit hardware timestamp */
+            std::uint64_t MB_timestamp = S32K::FlexCAN[instance]->RAMn[MB_index * S32K::MB_Size_Words] & 0xFFFF;
+
+            /* Instantiate monotonic object form a resolved timestamp */
+            time::Monotonic timestamp_ISR = resolve_Timestamp(MB_timestamp, instance);
+
+            /* Create Frame object with constructor */
+            CAN::Frame<CAN::TypeFD::MaxFrameSizeBytes> FrameISR(id_ISR,
+                                                                reinterpret_cast<std::uint8_t*>(data_ISR_word),
+                                                                dlc_ISR,
+                                                                timestamp_ISR);
+
+            /* Insert the frame into the queue */
+            S32K::g_frame_ISRbuffer[instance].push_back(FrameISR);
+        }
+        else
+        {
+            /* Increment the number of discarded frames due to full RX dequeue */
+            S32K::g_discarded_frames_count[instance]++;
+        }
+
+        /* Clear MB interrupt flag (write 1 to clear)*/
+        S32K::FlexCAN[instance]->IFLAG1 |= (1 << MB_index);
```

Comment:
```suggestion
        S32K::FlexCAN[instance]->IFLAG1 |= (1u << MB_index);
```

Reply:
Added in https://github.com/noxuz/platform_specific_components/commit/00e671a4dd917b2c71d324a2ca7b9202928f20fc